src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/CpsPackageReferenceProject.cs

```diff
@@ -116,6 +113,13 @@ private PackageSpec GetPackageSpec()
             return null;
         }
 
+        /// <inheritdoc/>
+        internal override async ValueTask<PackageSpec> GetPackageSpecAsync(CancellationToken ct)
+        {
+            ct.ThrowIfCancellationRequested();
+
+            return await Task.FromResult<PackageSpec>(GetPackageSpec());
```

Comment:
I just tried to 'force' a async implementation, so that `PackageReferenceProject` class has an abstract `GetPackageSpecAsync()`. Might need to add this as an abstract method and implement it in both project classes.

Reply:
Just to be clear, here's what I am recommending. 


```cs
internal override ValueTask<PackageSpec> GetPackageSpecAsync(CancellationToken ct)
{
            ct.ThrowIfCancellationRequested();
            return ValueTask.FromResult<PackageSpec>(GetPackageSpec());
}
```

You can't really force a method to be async, but rather you can only force it to return a task/valuetask. 

A method can always be implemented as sync, even if it returns a task.

Reply:
Changed to your implementation.